opal/corelib/basic_object.rb

```diff
@@ -34,6 +34,10 @@ def !
     false
   end
 
+  def !=(other)
+    self != other
```

Comment:
Why not implement it as follows?

```
def != other
  `self !== other`
end
```

That would avoid a stack overflow as it delegates directly to JavaScript. I could see, however, if this is a problem if Ruby and JavaScript don't agree on equality rules.